src/lib/upgradePackageDefinitions.ts

```diff
@@ -11,6 +11,77 @@ import { pickBy } from './pick'
 import queryVersions from './queryVersions'
 import upgradeDependencies from './upgradeDependencies'
 
+/**
+ * check if in peer violation
+ *
+ * @returns
+ */
+const checkIfInPeerViolation = (
+  currentDependencies: Index<VersionSpec>,
+  filteredUpgradedDependencies: Index<VersionSpec>,
+  upgradedPeerDependencies: Index<Index<VersionSpec>>,
+) => {
+  const upgradedDependencies = { ...currentDependencies, ...filteredUpgradedDependencies }
+  const upgradedDependenciesVersions = Object.fromEntries(
+    Object.entries(upgradedDependencies).map(([packageName, versionSpec]) => {
+      return [packageName, semver.minVersion(versionSpec)?.version ?? versionSpec]
+    }),
+  )
+  const filteredUpgradedPeerDependencies = { ...upgradedPeerDependencies }
+  let wereUpgradedDependenceFiltered = false
+  const filteredUpgradedDependenciesAfterPeers = pickBy(filteredUpgradedDependencies, (spec, dep) => {
+    const peerDeps = filteredUpgradedPeerDependencies[dep]
+    if (!peerDeps) {
+      return true
+    }
+    const valid = Object.entries(peerDeps).every(
+      ([peer, peerSpec]) =>
+        upgradedDependenciesVersions[peer] === undefined ||
+        semver.satisfies(upgradedDependenciesVersions[peer], peerSpec),
+    )
+    if (!valid) {
+      wereUpgradedDependenceFiltered = true
+      delete filteredUpgradedPeerDependencies[dep]
+    }
+    return valid
+  })
+  return {
+    issuesFound: wereUpgradedDependenceFiltered,
```

Comment:
Let's call this key `violated` for consistency.

Reply:
done